src/solvers/slepc_eigen_solver.C

```diff
@@ -743,6 +743,18 @@ void SlepcEigenSolver<T>::attach_deflation_space(NumericVector<T> & deflation_ve
 }
 
 template <typename T>
+void SlepcEigenSolver<T>::set_initial_space(NumericVector<T> & initial_space_in)
+{
+  this->init();
+  
+  PetscErrorCode ierr = 0;
+  Vec initial_vector = (cast_ptr<PetscVector<T> *>(&initial_space_in))->vec();
```

Comment:
(Just a reminder to self, this is not something for you to necessarily fix.)

I see that this idiom is used in several places in the `SlepcEigenSolver` code, but in optimized mode, `cast_ptr` simply does a `static_cast`, which could cause this line of code to segfault somewhat mysteriously if a non-PETSc `NumericVector` is passed in.  I'd like to change these to be done in two lines with a dynamic cast, checking the result.